packages/core/src/internal/fetchOptions.ts

```diff
@@ -55,18 +55,19 @@ export const makeFetchURL = (
     operation.kind === 'query' && operation.context.preferGetMethod;
   if (!useGETMethod || !body) return operation.context.url;
 
-  const url = new URL(operation.context.url);
+  const [url, params] = extractParams(operation.context.url);
   for (const key in body) {
     const value = body[key];
     if (value) {
-      url.searchParams.set(
+      params.set(
         key,
         typeof value === 'object' ? stringifyVariables(value) : value
       );
     }
   }
 
-  const finalUrl = url.toString();
+  const serialized = params.toString();
+  const finalUrl = serialized.length > 0 ? `${url}?${serialized}` : url;
```

Comment:
```suggestion
  const urlParts = extractParams(operation.context.url);
  for (const key in body) {
    const value = body[key];
    if (value) {
      urlParts[1].set(
        key,
        typeof value === 'object' ? stringifyVariables(value) : value
      );
    }
  }

  const finalUrl = urlParts.join('?');
```

nit: since we can assume that we'll always have search params we can do something a little fancier here ❤️ mostly we try to sometimes find these little changes to save on some bytes in the output bundle, just to add context on why I'm suggesting this